src/models/ssh.ts

```diff
@@ -199,19 +205,53 @@ export class SSH {
           if (!/\/$/.test(localFolderPath)) {
             localFolderPath += '/';
           }
-          const files = await fs.listTree(localFolderPath);
-
-          for (const localPath of files) {
-            const fileFolderPath = path
-              .dirname(localPath)
-              .replace(/[\/\\]+/g, '/');
-            const relativePath = fileFolderPath.substr(localFolderPath.length);
-            const remotePath = path.join(remoteFolderPath, relativePath);
-            await SSH.uploadFile(id, localPath, remotePath);
-          }
 
-          resolve();
-          return;
+          const client = SSH._getClient(id);
+
+          // Clean and create the target path if not exists.
+          const cmd = `rm -rf ${targetPath} && mkdir -p ${targetPath}`;
+          client.exec(cmd, (err: Error, channel: ssh2.ClientChannel) => {});
+
+          client.sftp(async (error: Error, sftp: ssh2.SFTPWrapper) => {
+            if (error) {
+              reject(error);
+              return;
+            }
+
+            const files = await fs.listTree(localFolderPath);
+
+            for (const filePath of files) {
+              const fileFolderPath = path
+                .dirname(filePath)
+                .replace(/[\/\\]+/g, '/');
```

Comment:
there are too many string replacement in code to convert path format from Windows to Unix/Linux. Can we refactor this logic to do the conversion at first, so that we don't need to convert again and again across the code?

In addition, I am not sure if "path" module already support the format between Windows and Unix/Linux so that we don't need to do manual string replacement at all?

Reply:
I will set up a task to track the path formatting logic refactor.

Reply:
You can use "path.win32" and "path.posix" to handling file path for Window and Unix/Linux respectively.

E.G. Insted using hard-code path seperator charactor everywhere, you can use the predefined const path.win32.sep to represent the '\' path seperator in windows; and use path.posix.sep to represent the '/' path seperator in Linux/Unix.

---
In reply to: [329335552](https://github.com/VSChina/vscode-iot-device-cube/pull/13#discussion_r329335552) [](ancestors = 329335552)

Reply:
Fixed.